### ISSUE TITLE
LFPT5: A UNIFIED FRAMEWORK FOR LIFELONG FEW-SHOT LANGUAGE LEARNING BASED ON PROMPT TUNING OF T5

### ISSUE THREAD

**Issue by HYUNMIN-HWANG** [One-Day-One-Paper/One-Day-One-Paper#287]
# LFPT5: A UNIFIED FRAMEWORK FOR LIFELONG FEW-SHOT LANGUAGE LEARNING BASED ON PROMPT TUNING OF T5

Lifelong Few-shot Language Learning (LFLL) 

일부 데이터(Few-shot)만 입력시켜서 generalize 시키고자 한다.

plus ) promt tuning을 더함 → LFPT5

---

두 단계로 나눌 수 있다. 

1. generate qseudo labeled samples of previous learned domains
2. combining with the new domain training data to alleviate catastrophic forgetting
- 그리고, 현재 도메인과 이전 도메인의 label consistency를 유지하기 위해서 minimizing KL divergence loss를 계산한다.
- 그리고. new task에 additional prompt embedding을 추가한다.

---

![Untitled](https://user-images.githubusercontent.com/70581043/190554551-12bc5ebd-c4dd-42ed-9052-8e0ae9d92131.png)

- 크게 두가지 경우로 나누어 실험했다
    1. 같은 task이지만 domain이 다른 경우
    2. 같은 domain이지만 task가 다른 경우

## Methodology : LFPT5
![Untitled 1](https://user-images.githubusercontent.com/70581043/190554564-0b8c84b5-2bdb-4a52-967d-b3f777e8c93a.png)

새로운 D_task와 pretrain model theta가 주어졌을 때의 pi를 구하는 것이 우리의 목표임 (우변의 첫번째 항)

- 이전 연구들의 방식 : fine tuning을 하거나 manually designed prompt 해서 parameter를 구했다. 하지만 overfitting의 문제 그리고 너무 많은 labor가 필요하다는 문제가 있음
- 해당 논문에서 제시한 것 : prompt embedding
    - freeae the pre-trained model theta
    - prepend a series of tunable tokens P
    - parameterized by pi
- a small number of prompt embedding으로 훈련할 수 있으며, training data에서 온 signal을 condense 하게 저장할 수 있다.

### Tasks

크게 3가지 tasks가 있다 : sequence labeling, text classification, text generatrion

이 모든 tasks를 text-to-text format으로 생각한다.

![Untitled 2](https://user-images.githubusercontent.com/70581043/190554583-ec5f3c26-2fab-4f6c-aa01-495bbe7270fe.png)

이전에는 memory-based method를 사용해서 훈련을 했는데 여기서는 외적인 memory가 필요가 없는 “task solver and a generator”를 제안했다. 

즉, 이전에 학습한 tasks들로 pseudo samples를 만들고, 현재 모델을 학습할 때 이전 학습한 지식을 refresh 시켜주는 방식

![새로운 도메인을 학습할 때 : 먼저 이전 도메인으로 수도 샘흘
![Untitled 3](https://user-images.githubusercontent.com/70581043/190554648-4e0b7477-1c00-4276-98ea-e9eda9616ba8.png)
새로운 도메인을 학습할 때 : 먼저 이전 도메인으로 수도 샘플을 만든 후, 새로운 학습 데이터와 합쳐서 학습을 진행한다. KL divergence loss를 사용해서 최적화를 시켜준다. | 새로운 태스크를 학습할 때 : 기존 embedding은 frozen 시키고 prompt embedding을 추가한다.

새로운 도메인을 학습할 때 : 먼저 이전 도메인으로 수도 샘흘을 만든 후, 새로운 학습 데이터와 합쳐서 학습을 진행한다. KL divergence loss를 사용해서 최적화를 시켜준다. | 새로운 태스크를 학습할 때 : 기존 embedding은 frozen 시키고 prompt embedding을 추가한다.

- task solver : input X를 읽은 후 output Y를 encode하는 방식
- data generator : 각기 다른 tasks와 domain에 해당하는 generation tokens를 만든다.
    

    
![Untitled 4](https://user-images.githubusercontent.com/70581043/190554701-2ee1cf51-be0c-4a3d-bf9d-5a66dbae5146.png)

### adapting to new domains

- generates pseudo samples of previous domains
- minimize  a KL divergence loss

### adapting to new task types

- new tasks에 해당하는 prompt token을 추가
- 이전 것은 frozen

### New domains
![Untitled 5](https://user-images.githubusercontent.com/70581043/190554717-4d726896-13a0-4126-84f3-b78d6c81cb7e.png)

- EWC, MAC
